backend/src/database/repositories/reportRepository.ts

```diff
@@ -199,34 +201,49 @@ class ReportRepository {
         const [start, end] = filter.createdAtRange
 
         if (start !== undefined && start !== null && start !== '') {
-          whereAnd.push({
+          advancedFilter.and.push({
             createdAt: {
-              [Op.gte]: start,
+              gte: start,
             },
           })
         }
 
         if (end !== undefined && end !== null && end !== '') {
-          whereAnd.push({
+          advancedFilter.and.push({
             createdAt: {
-              [Op.lte]: end,
+              lte: end,
             },
           })
         }
       }
     }
 
-    const where = { [Op.and]: whereAnd }
+    const parser = new QueryParser(
+      {
+        nestedFields: {
+          sentiment: 'sentiment.sentiment',
+        },
+      },
+      options,
+    )
```

Comment:
Because whenever we send a sentiment filter (without specifying which specific mood), we should be using `sentiment.sentiment` to filter. This makes the UX better, since API users only need to sent `sentiment`, not `sentiment.sentiment`. The rest is of the fields are because AWS is returning it, and it is impossible to map everything to 1 value without loosing information (3 dimensions vs 1).